### ISSUE TITLE
Using RevealingSplashView with tabbar and navigation controller

### ISSUE THREAD

**Issue by batuhanc** [PiXeL16/RevealingSplashView#6]
Hi,

My entry view is in a UINavigationController which is in a UITabBarController. When I add this to my entry view, the splash view eventually stays between the top navigation bar and bottom tabbar. I tried to set negative constraints on top and bottom, hoping to stretch the view but as the nature of tabbar and navigation bar is always to stay on top, this didn't work. I can maybe workaround this by having a top most view just for the splash view and initialize my tabs after the animation but do you know of any better way to accomplish this?

Thank you.

**Comment by PiXeL16** [comment]
Hi,
Not really, I pretty much had to go with the same route as you. 
Let me know if it works!

**Comment by batuhanc** [comment]
Hi,

I have ended up with creating a "splash" view controller and make it the root view controller in AppDelegate. After the animation ends, I transition to my actual tab bar container and set it as the root. This method works.

After I did that, I thought of another way which would be to hide the navigation bar and the tab bar until the splash view animation ends and show them again, but I didn't try that. Just noting for anyone with the same issue as this is much simpler to do if it works.

**Comment by PiXeL16** [comment]
Thanks for the information @batuhanc. Glad that it worked!

**Comment by MrBugBears** [comment]
Hey,

I tried the way to hide the navigation bar and show them after the animation with

``` swift
self.navigationController?.setNavigationBarHidden(false, animated: true)
```

Having two animations so short after another looked a bit strange. So I came up with a third option that works great for me. Instead of hiding/showing the navigation bar I don't add the RevealingSplashView to my current view. Instead I add it to the window itself with

``` swift
let window = UIApplication.sharedApplication().keyWindow
window?.addSubview(revealingSplashView)
```
